cli/command/trust/sign.go

```diff
@@ -79,9 +86,10 @@ func signImage(cli command.Cli, imageName string) error {
 	}
 	requestPrivilege := command.RegistryAuthenticationPrivilegedFunc(cli, imgRefAndAuth.RepoInfo().Index, "push")
 	target, err := createTarget(notaryRepo, tag)
-	if err != nil {
+	if err != nil || options.local {
 		switch err := err.(type) {
-		case client.ErrNoSuchTarget, client.ErrRepositoryNotExist:
+		// If the error is nil then the local flag is set
+		case client.ErrNoSuchTarget, client.ErrRepositoryNotExist, nil:
```

Comment:
can we have additional print output if the local flag was set and the image exists locally?  Something like:

`fmt.Fprintf(cli.Out(), "Signing and pushing trust metadata for locally tagged image %s, may overwrite remote trust data\n", imageName)`